src/components/overview/MacroList.tsx

```diff
@@ -1,27 +1,69 @@
 import { AddIcon } from '@chakra-ui/icons'
 import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
   Box,
   Button,
   Flex,
   Grid,
   GridItem,
+  HStack,
+  Text,
   useColorModeValue,
   VStack
 } from '@chakra-ui/react'
-import { useCallback } from 'react'
+import { useCallback, useMemo, useState } from 'react'
 import { useApplicationContext } from '../../contexts/applicationContext'
 import { useSelectedCollection } from '../../contexts/selectors'
 import { ViewState } from '../../constants/enums'
-import { Macro } from '../../types'
+import { Collection, Macro } from '../../types'
 import MacroCard from './MacroCard'
 import useScrollbarStyles from '../../hooks/useScrollbarStyles'
 import useMainBgColour from '../../hooks/useMainBgColour'
+import { motion } from 'framer-motion'
 
-export default function MacroList() {
-  const { selection, onCollectionUpdate, changeViewState } =
-    useApplicationContext()
+interface Props {
+  searchValue: string
+}
+
+export default function MacroList({ searchValue }: Props) {
+  // Get the ApplicationContext from your hook.
+  const {
+    collections,
+    selection,
+    onCollectionUpdate,
+    changeViewState,
+    isMacroOutputEnabled
+  } = useApplicationContext()
   const currentCollection = useSelectedCollection()
   const shadowColour = useColorModeValue('md', 'white-md')
+  const [matchingMacros, setMatchingMacros] = useState<
+    { macro: Macro; collection: Collection }[]
+  >([])
+
+  useMemo(() => {
+    const newMatchingMacros: { macro: Macro; collection: Collection }[] = []
+
+    if (searchValue) {
+      collections.map((collection) => {
+        collection.macros.map((macro) => {
+          if (macro.name.toLocaleLowerCase().includes(searchValue)) {
+            newMatchingMacros.push({ macro, collection })
+          }
+        })
+      })
+      setMatchingMacros(newMatchingMacros)
+    } else {
+      setMatchingMacros(
+        currentCollection.macros.map((macro, index) => ({
+          macro,
+          collection: currentCollection,
+          index
+        }))
+      )
+    }
+  }, [searchValue, collections, currentCollection])
```

Comment:
This can just be a useMemo, no need to be a useEffect

Reply:
Reworked: [d08d14a](https://github.com/WootingKb/wooting-macros/pull/194/commits/d08d14aebfef857760e22a9d9ef86a60410edc82)